salt/modules/osquery.py

```diff
@@ -0,0 +1,1288 @@
+        return text
+    return False
+
+
+def _osquery_cmd(table, attrs=None, where=None, format='json'):
+    '''
+    Helper function to run osquery queries
+    '''
+    if attrs:
+        if isinstance(attrs, list):
+            valid_attrs = _table_attrs(table)
+            if valid_attrs:
+                for a in attrs:
+                    if a not in valid_attrs:
+                        log.error('{0} is not a valid attribute for table {1}'.format(a, table))
+                        return False
+                _attrs = ','.join(attrs)
+            else:
+                log.error('Invalid table {0}.'.format(table))
+                return False
+        else:
+            log.error('attrs must be specified as a list.')
+            return False
+    else:
+        _attrs = '*'
+
+    sql = 'select {0} from {1}'.format(_attrs, table)
+
+    if where:
+        sql = '{0} where {1}'.format(sql, where)
+
+    sql = '{0};'.format(sql)
+
+    res = _osquery(sql)
+    if res:
+        return res
+    return False
+
+
+def version():
+    '''
+    Return version of osquery
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' osquery.version
+    '''
+    res = _osquery_cmd(table='osquery_info', attrs=['version'])
+    if res:
+        return res[0]['version']
+    return False
+
+
+def rpm_packages(attrs=None, where=None):
+    '''
+    Return cpuid information from osquery
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' osquery.rpm_packages
+    '''
+    if __grains__['os_family'] == 'RedHat':
+        res = _osquery_cmd(table='rpm_packages', attrs=attrs, where=where)
+        if res:
+            return res
+        return False
+    return False
+
+
+def kernel_integrity(attrs=None, where=None):
+    '''
+    Return kernel_integrity information from osquery
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' osquery.kernel_integrity
+    '''
+    if __grains__['os_family'] == 'RedHat' or __grains__['os_family'] == 'Debian':
+        res = _osquery_cmd(table='kernel_integrity', attrs=attrs, where=where)
+        if res:
+            return res
+        return False
+    return False
+
+
+def kernel_modules(attrs=None, where=None):
+    '''
+    Return kernel_modules information from osquery
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' osquery.kernel_modules
+    '''
+    if __grains__['os_family'] == 'RedHat' or __grains__['os_family'] == 'Debian':
+        res = _osquery_cmd(table='kernel_modules', attrs=attrs, where=where)
+        if res:
+            return res
+        return False
+    return False
+
+
+def memory_map(attrs=None, where=None):
+    '''
+    Return memory_map information from osquery
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' osquery.memory_map
+    '''
+    if __grains__['os_family'] == 'RedHat' or __grains__['os_family'] == 'Debian':
+        res = _osquery_cmd(table='memory_map', attrs=attrs, where=where)
```

Comment:
Why not then pass "supporting OS" right into `_osquery_cmd` but check it every time you call something? You could check all this stuff inside the `_osquery_cmd` instead.